File: WebSite/EulerAList/pset34.cpp
#include <iostream>   
#include <algorithm>    
#include <cmath>
#include <string>
#include <vector>
#include <unordered_set>

using namespace std;
unordered_set<int> res;


int main () 
{
    long result = 0;
    vector<long> valueList(10000000, 0);
    valueList[0] = 1;
    valueList[1] = 1;
    for (int i = 2; i <= 9; i ++)
    {
        valueList[i] = valueList[i - 1] * i;
    }
    for (int i = 1; i < 7; i ++)
    {
        unsigned long long start = pow(10, i);
        unsigned long long end = pow(10, i + 1);
        for (unsigned long long j = start; j < end; j ++)
        {
            valueList[j] = valueList[j / start] + valueList[j % start] + i - to_string(j % start).size();
            if (valueList[j] == j)
            {
                cout << j << ";";
                result += j;
            }
        }
    }
    cout << endl << result << endl;
}

File: WebSite/EulerAList/pset34.py
res = set()

result = 0
valueList = [0] * 10000000
valueList[0] = 1
valueList[1] = 1
for i in range(2, 10):
    valueList[i] = valueList[i - 1] * i

for i in range(1, 7):
    start = pow(10, i)
    end = pow(10, i + 1)
    for j in range(start, end):
        valueList[j] = valueList[j // start] + valueList[j % start] + i - len(str(j % start))
        if valueList[j] == j:
            print(j)
            result += j

print(result)
